Support introductoryOfferEligibilityJWS and promotionalOfferJWS in CUSTOM_ENTITLEMENT_COMPUTATION mode

fire-at-will opened the pull request:
<!-- Thank you for contributing to Purchases! Before pressing the "Create Pull Request" button, please provide the following: -->

### Description
This PR allows developers using the SDK in CEC mode to provide a `introductoryOfferEligibilityJWS` or an `offerID` & `promotionalOfferJWS` to apply to their StoreKit 2 purchases.

### Usage:
#### Introductory Offer Eligibility with JWS
```swift
let paramsWithIntroOfferJWS = PurchaseParams.Builder(product: storeProduct)
   .with(introductoryOfferEligibilityJWS: "abc")
   .build()

// Purchase the purchase params
```

#### Promotional Offers with JWS
```swift
let promoOfferPurchaseOptions = StoreKit2PromotionalOfferPurchaseOptions(
    offerID: "offer-id",
    compactJWS: "signed-jws"
)

let paramsWithPromoOfferOptions = PurchaseParams.Builder(product: storeProduct)
   .with(promotionalOfferOptions: promoOfferPurchaseOptions)
   .build()

// Purchase the purchase params
```

fire-at-will commented on Sources/Logging/Strings/StoreKitStrings.swift:
```diff
@@ -39,6 +39,10 @@ enum StoreKitStrings {
 
     case sk2_purchasing_added_winback_offer_option(String)
 
+    case sk2_purchasing_added_custom_introductory_offer_eligibility_jws
```
Explicitly not logging the JWS strings here to be extra cautious for security

fire-at-will commented:
@RCGitBot please test

tonidero reviewed (commented):
Looks good to me! Just a comment about hiding visibility of that new API but good otherwise

rickvdl commented on RevenueCat.xcodeproj/project.pbxproj:
```diff
@@ -1290,6 +1290,7 @@
 		F5FCD3EA27DA0D0B003BDC04 /* PriceFormatterProvider.swift in Sources */ = {isa = PBXBuildFile; fileRef = F5FCD3E927DA0D0B003BDC04 /* PriceFormatterProvider.swift */; };
 		F5FCD3FC27DA2034003BDC04 /* PriceFormatterProviderTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = F5FCD3FB27DA2034003BDC04 /* PriceFormatterProviderTests.swift */; };
+		FD05A71A2EE2430E00FE671F /* StoreKit2PromotionalOfferPurchaseOptions.swift in Sources */ = {isa = PBXBuildFile; fileRef = FD05A7192EE2430E00FE671F /* StoreKit2PromotionalOfferPurchaseOptions.swift */; };
 		FD15AF5B2DF9A8F30062467E /* VirtualCurrencyBalanceListRow.swift in Sources */ = {isa = PBXBuildFile; fileRef = FD15AF582DF9A8F30062467E /* VirtualCurrencyBalanceListRow.swift */; };
```
`StoreKit2PromotionalOfferPurchaseOptionsAPI.swift` is missing here, see https://github.com/RevenueCat/purchases-ios/pull/5908#issuecomment-3614571457 :)

JayShortway commented on Sources/Purchasing/StoreKit2/StoreKit2PromotionalOfferPurchaseOptions.swift:
```diff
@@ -0,0 +1,35 @@
+ * ``StoreKit2PromotionalOfferPurchaseOptions`` can be used to apply promotional offers to StoreKit 2 purchases.
+ */
+@objc(StoreKit2PromotionalOfferPurchaseOptions)
+public final class StoreKit2PromotionalOfferPurchaseOptions: NSObject, Sendable {
```
Yes we should!

fire-at-will replied:
Okay, made `StoreKit2PromotionalOfferPurchaseOptions` internal when not using CEC mode + added a detailed comment explaining the double class definitions + implementations in 79cf6185f0b224842f0c50da1354a2f1585aca04 😄

ajpallares commented on Sources/Purchasing/Purchases/PurchaseParams.swift:
```diff
@@ -140,6 +159,42 @@ import Foundation
+        // swiftlint:disable line_length
+        /**
+         * Sets an promotionalOffer JWS to be included with the purchase. StoreKit 2 only.
+         * - Parameter promotionalOfferJWS: The ``promotionalOffer JWS`` to apply to the purchase.
```
The parameter name is `promotionalOfferOptions` (and not `promotionalOfferJWS`). 

Also, the documentation of the parameter seems incorrect

ajpallares reviewed (commented):
Looking good! I have some more comments on documentation and on the public interface as well

fire-at-will commented:
@RCGitBot please test

fire-at-will commented:
Since y'all have last reviewed this PR, I've made the following changes:
- Made `StoreKit2PromotionalOfferPurchaseOptions` internal when not using CEC mode
- Removed the `StoreKit2PromotionalOfferPurchaseOptions` API tester file for non-CEC mode
- Named the `offerID` param in the promo offer logging enum case
- Updated a few of the public comments in the `PurchaseParams` class

Let me know if y'all see anything else 🙌
